ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java

```diff
@@ -46,16 +81,93 @@ public MessageInfo(StoreKey key, long size) {
    * Construct an instance of MessageInfo.
    * @param key the {@link StoreKey} associated with this message.
    * @param size the size of this message.
-   * @param deleted whethe the message is deleted.
+   * @param deleted {@code true} if the message is deleted, {@code false} otherwise
    * @param expirationTimeInMs the time at which the message will expire. A value of -1 means no expiration.
    * @param crc the crc associated with this message. If unavailable, pass in null.
    */
   public MessageInfo(StoreKey key, long size, boolean deleted, long expirationTimeInMs, Long crc) {
+    this(key, size, deleted, expirationTimeInMs, crc, ACCOUNTID_DEFAULT_VALUE, CONTAINERID_DEFAULT_VALUE,
+        Utils.Infinite_Time);
+  }
+
+  /**
+   * Construct an instance of MessageInfo.
+   * @param key the {@link StoreKey} associated with this message.
+   * @param size the size of this message.
+   * @param expirationTimeInMs the time at which the message will expire. A value of -1 means no expiration.
+   * @param accountId accountId of the blob
+   * @param containerId containerId of the blob
+   * @param operationTimeMs operation time in ms
+   */
+  public MessageInfo(StoreKey key, long size, long expirationTimeInMs, short accountId, short containerId,
```

Comment:
Concur with too many constructors - would it be ok to make the change as one commit (or another PR) so that reviewing is easier and quicker?